app/jobs/notify_web_hook_job.rb

```diff
@@ -48,44 +52,53 @@ def authorization
   end
 
   def hook_relay_url
-    "https://api.hookrelay.dev/hooks/#{ENV['HOOK_RELAY_ACCOUNT_ID']}/#{ENV['HOOK_RELAY_HOOK_ID']}/webhook_id-#{webhook.id}"
-  end
-
-  def use_hook_relay?
-    # can't use hook relay for `perform_now` (aka an unenqueued job)
-    # because then we won't actually hit the webhook URL synchronously
-    enqueued_at.present? && ENV["HOOK_RELAY_ACCOUNT_ID"].present? && ENV["HOOK_RELAY_HOOK_ID"].present?
+    "https://api.hookrelay.dev/hooks/#{account_id}/#{hook_id}/webhook_id-#{webhook.id || 'fire'}"
   end
 
   def post_hook_relay
     response = post(hook_relay_url)
-    delivery_id = JSON.parse(response.body).fetch("id")
+    delivery_id = response.body.fetch("id")
     Rails.logger.info do
       { webhook_id: webhook.id, url: webhook.url, delivery_id:, full_name: version.full_name, message: "Sent webhook to HookRelay" }
     end
+    return poll_delivery(delivery_id) if @poll_delivery
     true
   end
 
-  def post_directly
-    post(webhook.url)
-    true
-  rescue *ERRORS
-    webhook.increment! :failure_count
-    false
-  end
-
   def post(url)
-    Faraday.new(nil, request: { timeout: TIMEOUT_SEC }) do |f|
-      f.request :json
-      f.response :logger, logger, headers: false, errors: true
-      f.response :raise_error
-    end.post(
+    @http.post(
       url, payload,
       {
         "Authorization"   => authorization,
         "HR_TARGET_URL"   => webhook.url,
-        "HR_MAX_ATTEMPTS" => "3"
+        "HR_MAX_ATTEMPTS" => @poll_delivery ? "1" : "3"
       }
     )
   end
+
+  def poll_delivery(delivery_id)
+    deadline = (Rails.env.test? ? 0.01 : 10).seconds.from_now
+    response = nil
+    until Time.zone.now > deadline
+      sleep 0.5
```

Comment:
This seems like it could introduce a hung worker for up to 10 seconds for each job. Is there concern that this could be abused and cause delays on other queues?

Reply:
it will only be hit on the `fire` endpoint, and that was already a synchronous call to an arbitrary URL, so I think this is roughly equivalent